plugin.py

```diff
@@ -129,7 +185,168 @@ def run(self, edit: sublime.Edit) -> None:
         if session is None:
             return
 
-        session.send_request(Request("rust-analyzer/reloadWorkspace"))
+        session.send_request(Request("rust-analyzer/reloadWorkspace"), self.on_result)
+
+    def on_result(self, arg: Any):
+        pass
+
+
+
+class RustAnalyzerMemoryUsage(LspTextCommand):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+        session.send_request(Request("rust-analyzer/memoryUsage"), self.on_result)
+
+    def on_result(self, payload: str) -> None:
+        window = self.view.window()
+        if window is None:
+            return
+        sheets = window.selected_sheets()
+        view = window.new_file(flags=sublime.TRANSIENT)
+        view.set_scratch(True)
+        view.set_name("--- RustAnalyzer Memory Usage ---")
+        view.run_command("append", {"characters": "Per-query memory usage:"})
+        for line in payload.splitlines():
+            view.run_command("append", {"characters": '\n{}'.format(line)})
+        view.run_command("append", {"characters": "\n(note: database has been cleared)"})
+        view.set_read_only(True)
+        sheet = view.sheet()
+        if sheet is not None:
+            sheets.append(sheet)
+            window.select_sheets(sheets)
+
+
+RunnableArgs = TypedDict('RunnableArgs', {
+    'cargoArgs': List[str],
+    'cargoExtraArgs': List[str],
+    'executableArgs': List[str],
+    'overrideCargo': Optional[str],
+    'workspaceRoot': str,
+})
+Runnable = TypedDict('Runnable', {
+    'args': RunnableArgs,
+    'kind': str,
+    'label': str,
+})
+
+
+class RustAnalyzerExec(LspTextCommand):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        params = text_document_position_params(self.view, self.view.sel()[0].b)
```

Comment:
Style nit: Would make sense to create `params` just before `send_request`.

Reply:
Readability but I can change sure

Reply:
Not sure if we are talking about the same because IMO moving it a couple lines down wouldn't affect readability.
It would potentially affect unnecessary processing though (not really of any significance here but good practice nonetheless).